### PR TITLE
HPCC-12781 Add an ONFAIL attribute to FROMXML/FROMJSON

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review

File: ecl/regress/fromxml5err.ecl [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems.
```
Comment (jamienoss) [inline ecl/regress/fromxml5err.ecl]: 2015

File: testing/regress/ecl/fromxml5.ecl [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems.
```
Comment (jamienoss) [inline testing/regress/ecl/fromxml5.ecl]: 2015

File: ecl/regress/fromxml5err.ecl [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    ], { utf8 text; });
+
+contactRecord createFailure() := TRANSFORM
+    SELF.hasemail := TRUE;
+    SELF.email := FAILMESSAGE;
+    SELF := [];
+END;
+
+output(namesTable, { FROMXML(personRecord, text, TRIM, ONFAIL(createFailure())); });
+output(1);
```
Comment (jamienoss) [inline ecl/regress/fromxml5err.ecl]: needs newline before eof

File: testing/regress/ecl/fromxml5.ecl [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+personRecord createFailure() := TRANSFORM
+    SELF.surname := '**FAIL**';
+    SELF.forename := (UNICODE)FAILCODE;
+    SELF.contact.hasemail := TRUE;
+    SELF.contact.email := FAILMESSAGE;
+    SELF := [];
+END;
+
+output(namesTable, { FROMXML(personRecord, text, TRIM, ONFAIL(createFailure())); });
+output(1);
```
Comment (jamienoss) [inline testing/regress/ecl/fromxml5.ecl]: needs newline before eof

**Comment by jamienoss** [comment]
@richardkchapman good to merge, see comments and amend as you do so please, cheers.

**Comment by richardkchapman** [comment]
Merged manually to address Jamie's niggles.

**Pull request closed without merging**